modules/aerodyn/src/UnsteadyAero_Driver.f90

```diff
@@ -331,21 +335,29 @@ subroutine setUAinputs(n,u,t,dt,dvrInitInp,timeArr,AOAarr,Uarr,OmegaArr)
 
 
       else
-         indx = min(n,size(timeArr))
-         indx = max(1, indx) ! use constant data at initialization
+         ! check optional variables and allocation status
+         if (any( (/ allocated(timeArr),allocated(AOAarr),allocated(OmegaArr),allocated(Uarr) /) )) then
+             
+            indx = min(n,size(timeArr))
+            indx = max(1, indx) ! use constant data at initialization
          
-         ! Load timestep data from the time-series inputs which were previous read from input file
-         t       = timeArr(indx)
-         u%alpha = AOAarr(indx)*pi/180.0   ! This needs to be in radians
-         u%omega = OmegaArr(indx)
-         u%U     = Uarr(indx)
-         if (n> size(timeArr)) then
-            t = t + dt*(n - size(timeArr) ) ! update for NumInp>1;
-         elseif (n < 1) then
-            t = (n-1)*dt
+            ! Load timestep data from the time-series inputs which were previous read from input file
+            t       = timeArr(indx)
+            u%alpha = AOAarr(indx)*pi/180.0   ! This needs to be in radians
+            u%omega = OmegaArr(indx)
+            u%U     = Uarr(indx)
+            if (n> size(timeArr)) then
+              t = t + dt*(n - size(timeArr) ) ! update for NumInp>1;
+            elseif (n < 1) then
+              t = (n-1)*dt
+            end if
+            u%v_ac(1) = sin(u%alpha)*u%U
+            u%v_ac(2) = cos(u%alpha)*u%U
+         else
+            errStat = -1
```

Comment:
ErrStat should be set to `ErrID_Fatal` here. All of the error handling in OpenFAST assumes that errors are values > 0, and it's best practice to use the error level parameters.

Also, `ErrStat` and `ErrMsg` are not initialized in the other branches of the nested IF statement. So, this will result in unpredictable results on some compilers.